src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -161,6 +166,1134 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
+                    break;
+                case TYP_INT:
+                    ins = INS_pcmpgtd;
+                    break;
+                case TYP_BYTE:
+                    ins = INS_pcmpgtb;
+                    break;
+                case TYP_SHORT:
+                    ins = INS_pcmpgtw;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            if (ins != INS_cmppd)
+            {
+                emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op2Reg, TYP_SIMD16);
+            }
+            break;
+        }
+
+        case NI_SSE2_CompareGreaterThanOrEqual:
+        case NI_SSE2_CompareNotLessThan:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    ins = INS_cmppd;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            emit->emitIns_SIMD_R_R_R_I(ins, targetReg, op1Reg, op2Reg, 5, TYP_SIMD16);
+            break;
+        }
+
+        case NI_SSE2_CompareLessThan:
+        case NI_SSE2_CompareNotGreaterThanOrEqual:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    ins = INS_cmppd;
+                    break;
+                case TYP_INT:
+                    ins = INS_pcmpgtd;
+                    break;
+                case TYP_BYTE:
+                    ins = INS_pcmpgtb;
+                    break;
+                case TYP_SHORT:
+                    ins = INS_pcmpgtw;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            if (ins != INS_cmppd)
+            {
+                emit->emitIns_SIMD_R_R_R(ins, targetReg, op1Reg, op2Reg, TYP_SIMD16);
+            }
+            else
+            {
+                emit->emitIns_SIMD_R_R_R_I(ins, targetReg, op1Reg, op2Reg, 1, TYP_SIMD16);
+            }
+            break;
+        }
+
+        case NI_SSE2_CompareLessThanOrEqual:
+        case NI_SSE2_CompareNotGreaterThan:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                    ins = INS_cmppd;
+                    break;
+                default:
+                    unreached();
+                    break;
+            }
+
+            // Instruction with immediate byte operand controlling comparison type
+            emit->emitIns_SIMD_R_R_R_I(ins, targetReg, op1Reg, op2Reg, 2, TYP_SIMD16);
+            break;
+        }
+
+        case NI_SSE2_CompareNotEqual:
+        {
+            op2Reg = op2->gtRegNum;
+
+            instruction ins;
+            switch (baseType)
```

Comment:
You can drop the switch statement for the ones that only take a single type.

In the long run (for generic methods) we will want to be checking the types in the importer and returning a PNSE exception for the invalid cases.